WordPress/Classes/Services/MediaCoordinator.swift

```diff
@@ -156,16 +166,24 @@ class MediaCoordinator: NSObject {
     /// Retry the upload of a media object that previously has failed.
     ///
     /// - Parameter media: the media object to retry the upload
+    /// - Parameter isAutomaticRetry: indicates whether an upload is a manual action taken by the user
+    /// or an automagical one done via the `Uploader` protocol.
     ///
-    func retryMedia(_ media: Media, analyticsInfo: MediaAnalyticsInfo? = nil) {
+    func retryMedia(_ media: Media, analyticsInfo: MediaAnalyticsInfo? = nil, isAutomaticRetry: Bool = false) {
         guard media.remoteStatus == .failed else {
             DDLogError("Can't retry Media upload that hasn't failed. \(String(describing: media))")
             return
         }
 
         trackRetryUploadOf(media, analyticsInfo: analyticsInfo)
 
-        let coordinator = self.coordinator(for: media)
+        let coordinator: MediaProgressCoordinator
+        if isAutomaticRetry {
+            coordinator = autoRetryProgressCoordinator
```

Comment:
Two questions:

1. Is this change needed?  Why?
2. The original code went through a cache of existing media coordinators.  The new code doesn't - so there's a risk we may be changing logic that shouldn't be changed.